include/aws/common/atomics_msvc.inl

```diff
@@ -114,7 +143,8 @@ static inline void aws_atomic_priv_barrier_before(enum aws_memory_order order, e
      * Volatile ops may or may not imply this barrier, depending on the /volatile: switch, but adding an extra
      * barrier doesn't hurt.
      */
-    _ReadWriteBarrier();
+    RW_BARRIER();
```

Comment:
Could the combo be replaced by a single __dsb?

Reply:
the _ReadWriteBarrier is just a software barrier, it tells the compiler not to optimize around that line, so instructions jump across this barrier. but still the hardware can move commands around either by flipping or as a side effect of having the store buffer and the invalid queue.
dsb is a bit hard core and not needed in most cases (dmb is softer and works beautifully here)
since ARM has a weak memory model, a hardware barrier is needed.. i think we can get rid of the software barrier, but it has no effect here as the compiler doesn't issue any instruction for the `_ReadWriteBarrier()` similar to `asm volatile("" ::: "memory")` with gcc